### PR TITLE
Update SIDECAR_IMAGE tag in Helm chart variable

### LINKED ISSUE #3979: Add Helm Chart Variable for Configuring SIDECAR_IMAGE Independently
**Is your feature request related to a problem? Please describe.**
Currently, the SIDECAR_IMAGE environment variable can be set in the Agones controller, but it cannot be easily modified when installing Agones via Helm. This limitation forces the image registry to follow the common Agones settings, and it cannot be customized through deployment commands, arguments, or flags. For instance, if there is a need to use a different registry for the sidecar image, such as in cases where the sidecar image must be pulled from a private registry due to issues like those described in [issue #3978](https://github.com/googleforgames/agones/issues/3978), the current setup does not provide an easy way to achieve this.

**Describe the solution you'd like**
I would like to see the introduction of a Helm chart variable that allows the SIDECAR_IMAGE to be configured independently. This would enable users to specify a different image registry for the sidecar without affecting the main Agones configuration. It would provide more flexibility in managing image sources, especially in scenarios where different registries are required for different components.

**Describe alternatives you've considered**
The current alternative is to manually modify the deployment after the Helm installation or to maintain a custom fork of the Helm chart that includes the desired changes. However, these approaches are less efficient and more error-prone compared to having an official Helm variable for this purpose.

**Additional context**
This feature would be particularly useful in situations where the sidecar image needs to be pulled from a private registry, as in the case described in issue #3978. It would allow for easier maintenance and customization of Agones installations in environments with specific registry requirements.

### PR DESCRIPTION
**Pull request by 0xaravindh**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, press enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind breaking
> /kind bug

/kind cleanup

> /kind documentation
> /kind feature
> /kind hotfix
> /kind release

**What this PR does / Why we need it**:

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by peterzhongyi** [comment]
Issue #3979 mentioned the specific use case for adding a separate variable to specify the sidecar image is "useful in situations where the sidecar image needs to be pulled from a private registry". Can a image from a private registry be simply specified by a tag, or does it need some sort of repository url with a tag? In other words, how would pulling from a sidecar image a private registry be done with the tag?

**Comment by 0xaravindh** [comment]
We are currently facing an issue with specifying the [agones.sdk](https://pantheon.corp.google.com/artifacts/docker/agones-images/us/release/agones-sdk?mods=logs_tg_prod) tag  #3978. We are unable to specify the Windows tag and are only using the global tag.To address this, I have declared a specific tag for the SDK images. My question is whether we need to specify the entire URL, or if it would be sufficient to provide an option to declare the other Agones SDK tags? @peterzhongyi

**Comment by peterzhongyi** [comment]
@kangjk1017 Since you opened this issue, I thought it might be helpful to mention you here - do you think adding a variable to accept other Agones SDK tags will help fix the issue?

File: install/helm/agones/templates/controller.yaml [hunk 1]
```diff
@@ -123,7 +123,7 @@ spec:
           value: {{ .Values.gameservers.additionalPortRanges | toJson | quote }}
         {{- end }}
         - name: SIDECAR_IMAGE # overwrite the GameServer sidecar image that is used
-          value: "{{ .Values.agones.image.registry }}/{{ .Values.agones.image.sdk.name}}:{{ default .Values.agones.image.tag .Values.agones.image.sdk.tag }}"
```
[the remark this answers is not in the document]
Comment (0xaravindh) [reply install/helm/agones/templates/controller.yaml:126]: user has provided a comment regarding the same queries. (https://github.com/googleforgames/agones/issues/3979#issuecomment-2437423347)

[the remark this answers is not in the document]
Comment (gongmax) [reply install/helm/agones/templates/controller.yaml:126]: Yes, and what user asked "This is why I proposed the option to modify the registry for the SDK image." aligns with what I proposed above.

**Pull request closed without merging**